pkg/agent/runner.go

```diff
@@ -346,17 +354,24 @@ func (r *Runner) spawnBackgroundWorker(ctx context.Context, logger *zap.Logger,
 // getMetricsLoop repeatedly attempts to fetch metrics from the VM
 //
 // Every time metrics are successfully fetched, the value is recorded with newMetrics.
-func (r *Runner) getMetricsLoop(
+func getMetricsLoop[M core.FromPrometheus[C], C any](
+	r *Runner,
 	ctx context.Context,
 	logger *zap.Logger,
-	newMetrics func(metrics core.Metrics, withLock func()),
+	kind string,
```

Comment:
Can you have it as a part of an interface on C? 

Also, C is metrics names, right, maybe should be `Names`?

Reply:
> Can you have it as a part of an interface on C?

Not quite sure what you're referring to here — is it `kind`?

> Also, C is metrics names, right, maybe should be `Names`?

I was originally thinking we'd want some per-VM configuration for metrics for feature-gating new behavior, but upon reflection, I think implementing that properly is actually more effort than it's worth.

At the same time... I think the semantics are clearer if the interface defines it as "config" for metrics parsing, and then something external provides that "config" - even if it's always just the metric names in practice.

For now, I've added some comments to `FromPrometheus` (f47097a984e94f8b704a86f01d9c987e2dd1acd7). LMK what you think.

Reply:
I think 5dc98a95675fcb6573278834a688b123eda89149 satisfies this (in addition to the changes from [`4b06d09`..`0bb2b59`](https://github.com/neondatabase/autoscaling/compare/4b06d09cc3545f5f386cf1d8fe7bd90165ff7a75..0bb2b59f8326a70c6438a1623f98d69ae4c4f34e) ?